qunicorn_core/core/pilotmanager/qmware_pilot.py

```diff
@@ -205,7 +246,11 @@ def _get_job_results(self, qunicorn_job_id: int) -> None:  # noqa: C901
                         }
                     )
 
-                measurements: List[Dict] = json.loads(result["out"]["value"])
+                if program_state.data["batched"]:
+                    measurements: list[dict] = json.loads(result["out"]["value"])[program_state.data["circuit_index"]]
```

Comment:
If it is a batch result it would be better to try to save all results of that batch in one go. (i.e., we should only need to request and parse the batch result once). However, this will further complicate the code...
Maybe there can be a compromise between optimized code and added complexity, where we cache the batch result response and the parsed results for the duration of the loop?